### PR TITLE
Implement some missing v8 things

### PR DESCRIPTION
**Pull request by Yomanz**

### CODE REVIEW

File: data/src/main/scala/ackcord/data/message.scala [hunk 1]
```diff
@@ -441,7 +448,9 @@ case class GuildGatewayMessage(
     activity: Option[MessageActivity],
     application: Option[MessageApplication],
     messageReference: Option[MessageReference],
-    flags: Option[MessageFlags]
+    flags: Option[MessageFlags],
+    stickers: Option[List[Sticker]],
+    referencedMessage: Option[RawMessage] // TODO: Is this supposed to be RawMessage or GuildGatewayMessage
```
Comment (Katrix) [inline data/src/main/scala/ackcord/data/message.scala:453]: Unless you know that messages with references gotten with `REST` request also contain `GuildGatewayMessage`, then leave it as `Message`

Comment (Yomanz) [reply data/src/main/scala/ackcord/data/message.scala]: Yeah, I checked with advaith and he basically works for discord.

File: data/src/main/scala/ackcord/data/message.scala [hunk 2]
```diff
@@ -769,3 +778,25 @@ case class OutgoingEmbedAuthor(name: String, url: Option[String] = None, iconUrl
...
+  * @param packId id of the pack the sticker is from
+  * @param name name of the sticker
+  * @param description description of the sticker
+  * @param tags a comma-separated list of tags for the sticker
+  * @param asset sticker asset hash
+  * @param previewAsset sticker preview asset hash
+  * @param formatType	type of sticker format
+  */
+case class Sticker(
+    id: String,
```
Comment (Katrix) [inline data/src/main/scala/ackcord/data/message.scala:794]: Are these ids snowflakes? If so, just use `SnowflakeType[Sticker]` for the id. For packId, I'm not sure, so let's go with `RawSnowflake`.

File: data/src/main/scala/ackcord/data/message.scala [hunk 3]
```diff
@@ -769,3 +778,25 @@ case class OutgoingEmbedAuthor(name: String, url: Option[String] = None, iconUrl
...
+    packId: String,
+    name: String,
+    description: String,
+    tags: Option[String],
+    asset: String,
+    previewAsset: Option[String],
+    formatType: Int
```
Comment (Katrix) [inline data/src/main/scala/ackcord/data/message.scala:801]: Can you make this a proper enumeratum object like most other enums are?

**Review by Katrix — CHANGES_REQUESTED** [review]

File: data/src/main/scala/ackcord/data/message.scala (hunk 2 above)
Comment (Yomanz) [reply data/src/main/scala/ackcord/data/message.scala]: Implemented.

File: data/src/main/scala/ackcord/data/message.scala (hunk 3 above)
Comment (Yomanz) [reply data/src/main/scala/ackcord/data/message.scala]: Implemented.

File: data/src/main/scala/ackcord/data/message.scala [hunk 4]
```diff
@@ -66,6 +64,18 @@ object ImageData {
   }
 }
 
+sealed abstract class FormatType(val value: Int) extends IntEnumEntry
+object FormatType extends IntEnum[FormatType] with IntCirceEnumWithUnknown[FormatType] {
+  override def values: IndexedSeq[FormatType] = findValues
+
+  case object PNG extends FormatType(1)
+  case object APNG extends FormatType(2)
+  case object LOTTIE extends FormatType(3)
```
Comment (Yomanz) [inline data/src/main/scala/ackcord/data/message.scala:73]: I wasn't sure what case you wanted here, so I just did this because it looks nicer.

**Review by Katrix — CHANGES_REQUESTED** [review]
Build is failing on 2.12

**Comment by Yomanz** [comment]
> 
> 
> Build is failing on 2.12

I blame you :)

**Comment by Katrix** [comment]
Please don't touch TweetNacl

**Comment by Yomanz** [comment]
> 
> 
> Please don't touch TweetNacl

Consider them untouched

**Comment by Katrix** [comment]
There's still a diff for them

**Comment by Yomanz** [comment]
> 
> 
> There's still a diff for them

Consider them untouched^2

**Review by Katrix — APPROVED** [review]

**Pull request merged**